### PR TITLE
RESTEASY-1294: Fixed comparison bug in JaxrsInterceptorRegistry.

### PR DESCRIPTION
**Pull request by ronsigal**

### CODE REVIEW

File: jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/JaxrsInterceptorRegistry.java [hunk 1]
```diff
@@ -360,7 +360,15 @@ public JaxrsInterceptorRegistry(ResteasyProviderFactory providerFactory, Class<T
    {
       public int compare(Match match, Match match2)
       {
-         return match.order - match2.order;
```
Comment (JWGmeligMeyling) [inline jaxrs/resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/interception/JaxrsInterceptorRegistry.java]: This implementation is equal to `Integer.compareTo(match.order, match2.order);`

**Comment by liweinan** [comment]
@ronsigal @JWGmeligMeyling Can I merge this one?

**Comment by JWGmeligMeyling** [comment]
The new implementation is defenitely allright, but is already floating somewhere in the JRE. But changing that would be quite some hassle for a few lines less, so feel free to merge!

**Comment by ronsigal** [comment]
re: "Can I merge this one?"

Yeah. Not as pretty as Integer.compare(), but it's fine.

**Pull request merged**